### ISSUE TITLE
Help, the source code is missing some parts

### ISSUE THREAD

**Issue by nomanber** [kita0420/CFI-Net#1]
in [train_vessel.py] ，the 'from losses import  DiceBCELoss ' have No specific definition

**Comment by kita0420** [comment]
已收到您的邮件！

**Comment by kita0420** [comment]
In fact, this article does not use “DiceBCELoss”, but calculates BCELoss and DiceLoss separately. Reflected in line 110 of train_vessel.py.
The definition of DiceLoss is included in train_vessel.py, and&nbsp;the following is the definition of BCELoss.


def loss_ce(preds, masks, criterion, selected_mode = 1):

    '''
    :param preds:       N  H  W  C
    :param masks:       N  1  H  W
    :param criterion:   This is used to calculate nn.cross-entropy() or nn.BCE-loss(), both is OK !
    :return:            criterion (N*H*W, C  and  N,-1)
    '''
    if selected_mode == 1:       # when choose nn.BCELoss() masks should be float !
        masks = torch.tensor(masks, dtype=torch.float)
    outs = preds.permute((0, 2, 3, 1))         # N H W C
    outs = outs.reshape((-1, outs.size()[3]))  # N*H*W, C
    if selected_mode == 1:
        outs = outs.reshape((-1,))
    masks = masks.reshape((-1,))               # N,1,H,W ===&gt; N,-1
    if selected_mode == 2:
        masks = torch.tensor(masks, dtype=torch.long)
    return criterion(outs, masks)


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "kita0420/CFI-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年5月7日(星期二) 下午4:49
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[kita0420/CFI-Net] Help, the source code is missing some parts (Issue #1)





 
in [train_vessel.py] ，the 'from losses import  DiceBCELoss ' have No specific definition
